src/main.js

```diff
@@ -124,13 +137,15 @@ let load = base.loader({
   },
 
   api: {
-    requires: ['cfg', 'WorkerType', 'Secret', 'ec2', 'stateContainer', 'validator', 'publisher', 'influx'],
-    setup: async ({cfg, WorkerType, Secret, ec2, stateContainer, validator, publisher, influx}) => {
+    requires: ['cfg', 'WorkerType', 'AmiSet', 'Secret', 'ec2', 'validator', 'publisher', 'influx'],
+    setup: async ({cfg, WorkerType, AmiSet, Secret, ec2, validator, publisher, influx}) => {
+
       let reportInstanceStarted = series.instanceStarted.reporter(influx);
 
       let router = await v1.setup({
         context: {
           WorkerType: WorkerType,
+          AmiSet: AmiSet,
```

Comment:
Regarding my comment about `this.AmiSet.remove`, this is the place where you're creating the value `this.AmiSet`.  This is magic done by the library, but fyi.